### ISSUE TITLE
Jetpack / Polylang conflict -- RTL stylesheet issues

### ISSUE THREAD

**Issue by jeherve** [polylang/polylang#356]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [ ] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Steps to reproduce the issue

Hi there!

I am running into an annoying issue when both Jetpack and Polylang are active. This happens regardless of the theme in my testing, as long as the theme follows the following method for enqueuing the main stylesheet (just like Twenty Nineteen):

```php
wp_enqueue_style( 'twentynineteen-style', get_stylesheet_uri(), array(), wp_get_theme()->get( 'Version' ) );
wp_style_add_data( 'twentynineteen-style', 'rtl', 'replace' );
```

1. Activate both plugins.
2. Go to Languages > Settings > URL modifications, and pick this setting: The "language is set from content"
3. Load a page that should be using an RTL language, like Hebrew.
4. Note that the page is set to use that language: `<html dir="rtl" lang="he-IL">`
5. Yet the stylesheet that is loaded is the default one:
`<link rel="stylesheet" id="twentynineteen-style-css" href="http://yoursite.com/wp-content/themes/twentynineteen/style.css?ver=1.3" type="text/css" media="all">`
6. If you change the setting above, deativate Polylang, or deactivate Jetpack, the stylesheet is changed to the correct one:
`<link rel="stylesheet" id="twentynineteen-style-rtl-css" href="http://yoursite.com/wp-content/themes/twentynineteen/style-rtl.css?ver=1.3" type="text/css" media="all">`

I've tried to narrow things dow a bit on Jetpack's end but I can't seem to understand what could be the conflict here, so I thought I'd ask for your help. Does the option I mentioned above (`force_lang`) change anything with enqueues at any point? Does it use any specific WordPress hooks that I could look into and see if Jetpack uses them as well?

Thank you!

**Comment by Chouby** [comment]
It doesn't seem to be related to Jetpack. I can reproduce the bug consistently with only Twenty Nineteen. 

When the language is set by the content, the language is defined very late in an action hooked to `wp`. So it's required to update some WP global variables and, since I wasn't aware of this function `wp_style_add_data`, I missed `$wp_styles`.

**Comment by jeherve** [comment]
Thank you!

**Comment by yofon** [comment]
Hi JeHerve ! 
I am currently working on a website in French, English and Hebrew and it seems I have the same problem as yours. 
The hebrew mobile version doesn't display correctly in Hebrew with a big blank space that prevents the content to be seen (the content is deported to the right and the page seems empty as it is totally white). 
I am using Astra theme with Polylang. 
Did you find a solution to it ? 
Thanks for your reply. 
Kind regards,
Lionel
![WhatsApp Image 2020-12-10 at 11 26 27](https://user-images.githubusercontent.com/75936526/102017607-975cb980-3d70-11eb-8d73-7b9736f1296a.jpeg)

**Comment by jeherve** [comment]
@yofon The issue I was running into was fixed in Polylang 2.5.4. If you still experience issues with a newer version of the plugin, I would recommend that you contact Polylang support or create a new issue with as many details as possible about your setup.

Before you do that, I would recommend switching to one of the default WordPress themes for a few minutes, to ensure that the problem isn't limited to your current theme.

**Comment by yofon** [comment]
Hi, thanks for your reply.
It did not work. It seems that Polylang is the cause of this bug in hebrew
as it remains when changing template.
I took it off and try some other free plugins.
As it did not work or they were too complicated to install, I abandonned
them and only used a conditional menu to adapt each page to the right
language.
I think i just need to change the language in meta description. Would you
know how to do it in a easy way, only fo some specific pages ?
Thanks for your reply.
Lionel

Le lun. 14 déc. 2020 à 10:13, Jeremy Herve <notifications@github.com> a
écrit :
